examples/tts/magpietts_inference.py

```diff
@@ -0,0 +1,603 @@
+# Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+MagpieTTS Inference and Evaluation Script.
+
+This script provides a clean CLI for running MagpieTTS inference with optional evaluation.
+It decouples inference and evaluation into separate modules for better maintainability.
+
+Example usage:
+    # Inference only (from .nemo file) - default behavior
+    python examples/tts/magpietts_inference.py \\
+        --nemo_files /path/to/model.nemo \\
+        --datasets libritts_test_clean \\
+        --out_dir /path/to/output \\
+        --codecmodel_path /path/to/codec.nemo
+
+    # Inference with evaluation (from checkpoint)
+    python examples/tts/magpietts_inference.py \\
+        --hparams_files /path/to/hparams.yaml \\
+        --checkpoint_files /path/to/model.ckpt \\
+        --datasets libritts_test_clean,vctk \\
+        --out_dir /path/to/output \\
+        --codecmodel_path /path/to/codec.nemo \\
+        --run_evaluation \\
+        --num_repeats 3
+"""
+from __future__ import annotations
+
+import argparse
+import copy
+import json
+import logging
+import os
+import shutil
+from pathlib import Path
+from typing import List, Optional, Tuple
+
+# Import dataset configuration
+import examples.tts.magpietts.evalset_config as evalset_config
+import numpy as np
+
+# Import the modular components
+from examples.tts.magpietts.evaluation import (
+    DEFAULT_VIOLIN_METRICS,
+    STANDARD_METRIC_KEYS,
+    EvaluationConfig,
+    compute_mean_with_confidence_interval,
+    evaluate_generated_audio_batch,
+)
+from examples.tts.magpietts.inference import InferenceConfig, MagpieInferenceRunner
+from examples.tts.magpietts.utils import ModelLoadConfig, get_experiment_name_from_checkpoint_path, load_magpie_model
```

Comment:
better not to nest `utils` inside `examples.tts.magpeitts`. I would suggest move to `nemo/collections/tts/parts/utils/eval_utils.py` similar to what ASR did as in `nemo/collections/asr/parts/utils/eval_utils.py`